lib/vagrant-service-manager/service.rb

```diff
@@ -16,8 +16,13 @@ def initialize(app, env)
 
       def call(env)
         @app.call(env)
-        return false unless SUPPORTED_BOXES.include? @machine.guest.capability(:os_variant)
-        @service_hooks.each(&:execute)
+
+        if SUPPORTED_BOXES.include? @machine.guest.capability(:os_variant)
+          return false unless SUPPORTED_BOXES.include? @machine.guest.capability(:os_variant)
+          @service_hooks.each(&:execute)
+        end
+      rescue Vagrant::Errors::GuestCapabilityNotFound
+        # Do nothing if supported box variant not found
```

Comment:
+1 for debug log as users who are not using ADB/CDK would not bother about VSM. But debug log will help in case user is trying to make VSM work with anything other than ADB/CDK.